### PR TITLE
Randomise results

### PR DESCRIPTION
**Pull request by akenzua**
[TP11072](https://maps.tpondemand.com/restui/board.aspx?#page=board/5682182231901749200&appConfig=eyJhY2lkIjoiNzlGODk3MENDQkM0N0FCM0ZFQzkyNEVFODEyREM0M0EifQ==&boardPopup=userstory/11072/silent)

This PR randomised the results displayed every time its being filtered. Though out of the scope of this PR, it also fix a bug in the pagination and corrected the name collision in the translation file.

### CODE REVIEW

**Comment by davidtrussler** [comment]
I think the intention of this work needs some clarification before we can review this PR. My understanding of the task from reading the TP ticket is to randomise the results only once: on the initial load of firms. We should then produce a subset of that array each time the user makes a filter selection. The work in this PR appears to randomise the result on each instance of the user making a selection.

**Review by ladamsatmaps19 — APPROVED** [review]

**Comment by akenzua** [comment]
> I think the intention of this work needs some clarification before we can review this PR. My understanding of the task from reading the TP ticket is to randomise the results only once: on the initial load of firms. We should then produce a subset of that array each time the user makes a filter selection. The work in this PR appears to randomise the result on each instance of the user making a selection.

Clarified this with Dobrin. It now randomise results once on initial load. Sorted. Thanks.

File: pages/listings.js [hunk 1]
```diff
@@ -43,6 +44,8 @@ const Listings = ({ t }) => {
 Listings.getInitialProps = async ({ reduxStore }) => {
   await reduxStore.dispatch(getAlgoFirms());
   await reduxStore.dispatch(getAlgoOfferings());
+  const shuffledfirm = await reduxStore.getState().listings.firms.hits;
+  await randomizeResult(shuffledfirm);
```
Comment (davidtrussler) [inline pages/listings.js:48]: Minor point: we probably don't need `await` here as this isn't asynchronous. Would it be slightly more efficient to leave it out? I dunno. WDYT? 🤔

**Comment by davidtrussler** [comment]
There is a comment in the TP card "Also, whenever user switches between result pages, the browser should scroll up to the top of the page." which appears not to be covered here. Though this seems a little out of scope for this card we should ensure it is covered somehow. 

Otherwise I left one small comment but everything else looks good!

File: pages/listings.js (hunk 1 above)
Comment (akenzua) [reply pages/listings.js:48]: The randonmize function result takes a while to resolve and sometimes return undefined, so the await there is to ensure it returns result before the other code executes. Just not taking chances with it.

**Comment by akenzua** [comment]
> There is a comment in the TP card "Also, whenever user switches between result pages, the browser should scroll up to the top of the page." which appears not to be covered here. Though this seems a little out of scope for this card we should ensure it is covered somehow.
> 
> Otherwise I left one small comment but everything else looks good!

Sorted!

**Comment by davidtrussler** [comment]
> > There is a comment in the TP card "Also, whenever user switches between result pages, the browser should scroll up to the top of the page." which appears not to be covered here. Though this seems a little out of scope for this card we should ensure it is covered somehow.
> > Otherwise I left one small comment but everything else looks good!
> 
> Sorted!

The scroll should be on moving between pages. I think this is on the filter event.

**Comment by akenzua** [comment]
> > > There is a comment in the TP card "Also, whenever user switches between result pages, the browser should scroll up to the top of the page." which appears not to be covered here. Though this seems a little out of scope for this card we should ensure it is covered somehow.
> > > Otherwise I left one small comment but everything else looks good!
> > 
> > 
> > Sorted!
> 
> The scroll should be on moving between pages. I think this is on the filter event.

Yeah! Its been added

File: components/listingsPage/results.js [hunk 2]
```diff
@@ -132,8 +132,14 @@ const Results = ({ t }) => {
...
-          nextClick={() => setCurrentPage(currentPage + 1)}
-          prevClick={() => setCurrentPage(currentPage - 1)}
+          nextClick={() => {
+            setCurrentPage(currentPage + 1);
+            window.scrollTo(0, 0);
+          }}
+          prevClick={() => {
+            setCurrentPage(currentPage - 1);
+            window.scrollTo(0, 0);
+          }}
```
[the remark this answers is not in the document]
Comment (akenzua) [reply components/listingsPage/results.js:141]: > 👍 Looks good - but can we also remove the scroll function added to the filter event? As far as I understand we only need it here.

Removed

**Review by davidtrussler — APPROVED** [review]
LGTM - Nice work! 👍

**Pull request merged**